storage/spanstore/interface.go

```diff
@@ -36,7 +36,7 @@ type Reader interface {
 	// GetTrace retrieves the trace with a given id.
 	//
 	// If no spans are stored for this trace, it returns ErrTraceNotFound.
-	GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error)
+	GetTrace(ctx context.Context, query *TraceIDQueryParameters) (*model.Trace, error)
```

Comment:
```suggestion
	GetTrace(ctx context.Context, query TraceIDQueryParameters) (*model.Trace, error)
```
The query is required, no reason to pass via pointer